sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Filter.java

```diff
@@ -207,43 +166,54 @@ public void populateDisplayData(DisplayData.Builder builder) {
    * <p>See also {@link #byPredicate}, which returns elements
    * that satisfy the given predicate.
    */
-  public static <T extends Comparable<T>> ParDo.Bound<T, T> greaterThanEq(final T value) {
-    return ParDo.named("Filter.greaterThanEq").of(new DoFn<T, T>() {
-      @Override
-      public void processElement(ProcessContext c) {
-        if (c.element().compareTo(value) >= 0) {
-          c.output(c.element());
-        }
-      }
-
+  public static <T extends Comparable<T>> Filter<T> greaterThanEq(final T value) {
+    return byPredicate(new SerializableFunction<T, Boolean>() {
       @Override
-      public void populateDisplayData(DisplayData.Builder builder) {
-        super.populateDisplayData(builder);
-        Filter.populateDisplayData(builder, String.format("x ≥ %s", value));
+      public Boolean apply(T input) {
+        return input.compareTo(value) >= 0;
       }
-    });
+    }).named("Filter.greaterThanEq").described(String.format("x ≥ %s", value));
   }
 
   ///////////////////////////////////////////////////////////////////////////////
 
+  private String name = "Filter";
```

Comment:
Leave as `super(name)` or just remove it, as before. We want to get rid of `named` and if the only place it is stored is on the base `PTransform`, then we can remove it there and let the type system guide us through the places that call it.

(incidentally, if this weren't the plan I would favor your approach of minimizing communication up and down the inheritance hierarchy)